app/src/main/java/com/andrew/tasky/presentation/agenda/AgendaViewModel.kt

```diff
@@ -31,11 +32,47 @@ class AgendaViewModel@Inject constructor(
 
     fun setDateSelected(dateUserSelected: LocalDate) {
         _dateSelected.value = dateUserSelected
+        _calendarDateItemList.value.map { item ->
+            item.isSelected = dateUserSelected == item.date
+        }
     }
 
+    private val _calendarDateItemList = MutableStateFlow(
+        listOf<CalendarDateItem>(
+            CalendarDateItem(
+                true,
+                LocalDate.now()
+            )
+        )
+    )
+    val calendarDateItemList = _calendarDateItemList.asStateFlow()
+
     fun deleteAgendaItem(agendaItem: AgendaItem) {
         viewModelScope.launch {
             repository.deleteAgendaItem(agendaItem)
         }
     }
+
+    init {
+        val daysBefore = 6
+        val daysAfter = 8
+
+        for (day in 1..daysBefore) {
+            _calendarDateItemList.value += CalendarDateItem(
+                false,
+                LocalDate.now().minusDays(day.toLong())
+            )
+        }
+        for (day in 1..daysAfter) {
+            _calendarDateItemList.value += CalendarDateItem(
+                false,
+                LocalDate.now().plusDays(
+                    day.toLong()
+                )
+            )
+        }
```

Comment:
I'd first calculate the final list you want to assign and then assign it, since this way, you'll trigger many UI updates (with every call on `+=`)